File: src/main/java/com/github/kindrat/cassandra/client/ui/window/menu/KeySpaceProvider.java
package com.github.kindrat.cassandra.client.ui.window.menu;

import java.util.List;
import java.util.concurrent.CompletableFuture;

@FunctionalInterface
public interface KeySpaceProvider {
    CompletableFuture<List<String>> loadKeyspaces(String url, String username, String password);
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/ui/window/menu/KeySpaceProvider.kt
package com.github.kindrat.cassandra.client.ui.window.menu

import java.util.concurrent.CompletableFuture

@FunctionalInterface
fun interface KeySpaceProvider {
    fun loadKeyspaces(url: String, dc: String, username: String?, password: String?): CompletableFuture<List<String>>
}